src/calls.c

```diff
@@ -2006,17 +1993,17 @@ Obj FuncHandlerCookieOfFunction(Obj self, Obj func)
 */
 void SaveFunction ( Obj func )
 {
-  UInt i;
-  for (i = 0; i <= 7; i++)
-    SaveHandler(HDLR_FUNC(func,i));
-  SaveSubObj(NAME_FUNC(func));
```

Comment:
We cannot use `SET_NAME_FUNC` etc. during loading, because of the assertion in them -- during loading, we see incomplete objects which can (and does) trigger the assertions.

Saving/loading is handled by GASMAN, and I am pretty sure no GC can occur during it -- no code should call NewBag during loading anyway (I am not sure we enforce that, though... might be a good idea to add a global flag indicating that we are loading / saving a workspace, and if then something calls NewBag, we `abort()`. Similar during a GC (e.g. a mark function calling `NewBag` should lead to an `abort()`, not some weird hard to debug error)